crates/api/src/local_user/change_password_after_reset.rs

```diff
@@ -33,11 +33,16 @@ impl Perform for PasswordChangeAfterReset {
       return Err(LemmyError::from_message("passwords_dont_match"));
     }
 
+    // Expire reset token
+    // TODO do this in a transaction along with the user update (blocked by https://github.com/LemmyNet/lemmy/issues/1161)
+    PasswordResetRequest::expire(context.pool(), reset_request.id).await?;
```

Comment:
Transactions should be doable now that https://github.com/LemmyNet/lemmy/pull/3420 is merged.